### ISSUE TITLE
`EnablePlugin` and `DisablePlugin` types broken

### ISSUE THREAD

**Issue by squidsoup** [DavidWells/analytics#218]
The implementation of `analytics.plugins.[enable|disable]` suggests that the correct type should be:

```
 (plugins: string | string[], callback: (...params: any[]) => any) => Promise<any>
```
but instead, the type of the plugins arg is:

```
string & any[]
```
which means `(string) & (any[])` which is clearly not the intention, and breaks any calling code in strict mode.

**Comment by DavidWells** [comment]
Hey @squidsoup 

Good catch.

Should the resulting type look like this?

```ts
type EnablePlugin = (plugins: string & string[], callback?: (...params: any[]) => any) => Promise<any>;

type DisablePlugin = (plugins: string & string[], callback: (...params: any[]) => any) => Promise<any>;
```

**Comment by DavidWells** [comment]
Or is it 

```ts
type EnablePlugin = (plugins: string | string[], callback?: (...params: any[]) => any) => Promise<any>;
type DisablePlugin = (plugins: string | string[], callback: (...params: any[]) => any) => Promise<any>;
```

**Comment by squidsoup** [comment]
I believe it should be `string | string[]`, i.e. plugins should either be a string or array of strings, here's [an example in the ts playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBA6glsAFgeQE5QLxQM7FXAOwHMoAfHPQogbQF0BuAWAChRJYFEBBAgE0wr5iUAGSCqdJsxYsAxgHsCuKAEMAXByRoBAch1SFS4FABGG+FvRZqOgGbz5OgDRQdJlah20ZzQ8tnmnDz8WHoGisq8gUjBAjb2ji5uHl5AA).

**Comment by DavidWells** [comment]
Updated via https://github.com/DavidWells/analytics/pull/219

Released in `analytics@0.7.15`

Can you give those new types a test? 😃

**Comment by wgwz** [comment]
@DavidWells sorry necrobump an old issue but should this:
```
type DisablePlugin = (plugins: string | string[], callback: (...params: any[]) => any) => Promise<any>;
```
be this:
```
type DisablePlugin = (plugins: string | string[], callback?: (...params: any[]) => any) => Promise<any>;
```
note: the `?` such that the type signature is the same for `EnablePlugin`.

i'm assuming that just like you do this to enable a plugin:

```
/* User opted in, enable analytics plugins */
analytics.plugins.enable(['google-analytics', 'hubspot']).then(() => {
  /* Plugins are now loaded into the page */

  /* Track a page view */
  analytics.page()

  /* Track a custom event */
  analytics.track('cartCheckout', {
    item: 'pink socks',
    price: 20
  })

  /* Identify a visitor */
  analytics.identify('userId123', {
    name: 'bob',
    email: 'bob@bob.com'
  })
})
```

you would do this to disable:

```
analytics.plugins.disable(['google-analytics', 'hubspot']).then(() => {
  /* Plugins are now loaded into the page */

  /* Track a page view */
  // noop
  analytics.page()

  /* Track a custom event */
  // noop
  analytics.track('cartCheckout', {
    item: 'pink socks',
    price: 20
  })

  /* Identify a visitor */
  // noop
  analytics.identify('userId123', {
    name: 'bob',
    email: 'bob@bob.com'
  })
})
```
